### PR TITLE
Fix a long-standing bug for scoped access to internal modules.

### PR DESCRIPTION
**Pull request by mppf**
See also PR #4476 and PR #4517 which reverted it.  This PR makes a simple change to findVisibleFunctions to correctly handle the case when a scoped access to a module provided automatically does not compile; for example 

``` chapel
  writeln(List.makeList(1,2,3));
```

(with or without a 'use List').

findVisibleFunctions contains an optimization which is important for the performance of the compiler. This optimization is to store all included-by-default symbols into one scope instead of performing extra bookkeeping for these scopes across all of the included-by-default code.  The code handling List.makeList (e.g.) simply needed to be updated to use this one scope when appropriate. 

Passed full local testing.
Reviewed by @lydia-duncan - thanks!

### CODE REVIEW

**Comment by lydia-duncan** [comment]
The second sentence in your first paragraph has a typo but I'm not sure how to fix it.

File: compiler/resolution/visibleFunctions.cpp [hunk 1]
```diff
@@ -31,6 +31,24 @@
...
+
+   In this way, the symbols declared in each block are stored.
+   These tables do not save the transitive symbols available.
+   To do that, code using the table needs to go up blocks/scopes
+   and explicitly consider module 'use's.
+
+   This file also includes an optimization, where the
+   symbols available to all modules (i.e. what is in ChapelStandard)
+   is considered to be in a single block. This optimization
+   provides a significant performance for compiling 'hello'.
```
Comment (lydia-duncan) [inline compiler/resolution/visibleFunctions.cpp]: Missing the word "improvement" here?

**Comment by mppf** [comment]
In the PR message? I think I fixed it.

**Comment by mppf** [comment]
I don't expect any impact on compiler performance (but havn't looked at it in detail). I'll fix the missing word in the comment.

**Pull request merged**